### PR TITLE
Unit testing & soft refactor

### PR DESCRIPTION
**Pull request by lucasmenendez**
This PR attempts to solve #22 and #30

### CODE REVIEW

**Review by jordipainan — APPROVED** [review]

File: census/census_test.go [hunk 1]
```diff
@@ -2,17 +2,22 @@ package census
...
 	"github.com/ethereum/go-ethereum/common"
+	qt "github.com/frankban/quicktest"
 	"go.vocdoni.io/dvote/api/censusdb"
 	"go.vocdoni.io/dvote/censustree"
 	"go.vocdoni.io/dvote/data/compressor"
+	"go.vocdoni.io/proto/build/go/models"
 )
 
-var internalAddresses = map[common.Address]*big.Int{
+var MonkeysAddresses = map[common.Address]*big.Int{
```
Comment (jordipainan) [inline census/census_test.go:20]: what is `MonkeysAddresses` ?

File: census/census_test.go [hunk 2]
```diff
@@ -24,51 +29,162 @@ var internalAddresses = map[common.Address]*big.Int{
...
+	c.Assert(err, qt.ErrorIs, ErrCreatingCensusDB)
+
 	cdb, err := NewCensusDB(t.TempDir(), "")
-	if err != nil {
-		t.Fatal(err)
-	}
+	c.Assert(err, qt.IsNil)
+	c.Assert(cdb.ipfsConn, qt.IsNil)
+	err = cdb.storage.Stop()
+	c.Assert(err, qt.IsNil)
```
Comment (jordipainan) [inline census/census_test.go:42]: you can `c.Assert(cdb.storage.Stop(), ...)`

File: census/census_test.go [hunk 3]
```diff
@@ -24,51 +29,162 @@ var internalAddresses = map[common.Address]*big.Int{
...
 		tval, _, err := tree.GenProof(key[:censustree.DefaultMaxKeyLen])
-		if err != nil {
-			t.Fatal(err)
-		}
+		c.Assert(err, qt.IsNil)
+		c.Assert(tval, qt.ContentEquals, tree.BigIntToBytes(val))
+	}
+}
+
+func Test_newTree(t *testing.T) {
```
Comment (jordipainan) [inline census/census_test.go:88]: We do normally use camel, so `TestNewTree`

File: census/census_test.go [hunk 4]
```diff
@@ -24,51 +29,162 @@ var internalAddresses = map[common.Address]*big.Int{
...
+	c := qt.New(t)
+	cdb, err := NewCensusDB(t.TempDir(), "")
+	c.Assert(err, qt.IsNil)
+	defer func() {
+		err = cdb.storage.Stop()
```
Comment (jordipainan) [inline census/census_test.go:93]: One line: `c.Assert(cdb.storage)...`

File: census/census_test.go (hunk 1 above)
Comment (lucasmenendez) [reply census/census_test.go:20]: the holder addresses of our test token called "Monkeys"

**Review by jordipainan — COMMENTED** [review]
Some of the comments can be applied in other lines than the reported. If you are in a hurry feel free to ignore them as the functionality is not changed. But I think is nice to be as polite as we can in these things.

Thank you for the work :)

File: census/census_test.go (hunk 1 above)
Comment (jordipainan) [reply census/census_test.go:20]: Let the naming battle begin :P

File: census/census_test.go (hunk 4 above)
Comment (lucasmenendez) [reply census/census_test.go:93]: all occurrences solved!

File: census/census_test.go (hunk 1 above)
Comment (lucasmenendez) [reply census/census_test.go:20]: TokenHolders is already the name of a struct :(

File: census/census_test.go (hunk 3 above)
Comment (jordipainan) [reply census/census_test.go:82]: sorry i did not think that this was made in purpose for not exposing functions

**Pull request merged**